model/src/lif.py

```diff
@@ -0,0 +1,43 @@
+from typing import Any, Optional
+
+import torch
+import torch.nn as nn
+
+
+class LIF(nn.Module):
+    def __init__(self, beta: float, threshold: float = 1.0):
+        """
+        Note: Initialization of mem at initial forward pass will cause
+        transience in execution
+        """
+        super(LIF, self).__init__()
+        # Initialize decay rate beta and threshold
+        self.beta = beta
+        self.threshold = threshold
+        self.spike_op = self.SpikeOperator.apply
+        self.mem: Optional[torch.Tensor] = None
+
+    def forward(self, current: torch.Tensor) -> torch.Tensor:
+        if self.mem == None:  # noqa
+            self.mem = torch.zeros_like(current)
+
+        spk: torch.Tensor = self.spike_op(self.mem, self.threshold)
+        reset = spk * self.threshold
+        self.mem = self.beta * self.mem + current - reset  # type: ignore [operator]
```

Comment:
If you want to incorporate the negative resting potential we need to add the 
`self.mem-self.Vr` replacement of the self.mem in line 26. That may be a future pull?

Reply:
I think the open question is how much we want to mimic the implementation of LPL on top of Auryn. If we go 100% alignment we will implement axonal delays. If we go 75% we will foregoe axonal delays and anything more complicated, but include simple stuff like this -50mV resting membrane. If we go 50%, we will implement the LPL rule without resting membrane or axonal delays.

(percentages are arbitrary)

I'm down to follow your judgement here. Will add a note to followup.